src/app/Signup/page.tsx

```diff
@@ -0,0 +1,176 @@
+import Link from 'next/link';
+
+const SignupPage: React.FC = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmpassword, setconfirmPassword] = useState('');
+  const [emailError, EsetError] = useState(' ');
+  const [passError, PsetError] = useState(' ');
+  const [ConfirmpassError, confirmPsetError] = useState(' ');
+  const [shakeEmail, setShakeEmail] = useState(false);
+  const [shakePassword, setShakePassword] = useState(false);
+  const [shakeConfirmPass, setShakeConfirmPassword] = useState(false);
+
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setEmail(value);
+  };
+
+  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setPassword(value);
+  };
+  const handleConfirmPasswordChange = (
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const value = e.target.value;
+    setconfirmPassword(value);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    EsetError('');
+    PsetError('');
+    confirmPsetError('');
+    setShakeEmail(false);
+    setShakePassword(false);
+    setShakeConfirmPassword(false);
+
+    let valid = true;
+
+    // Validate email
+    if (!email) {
+      EsetError('Email cannot be empty.');
+      setShakeEmail(true);
+      valid = false;
+    } else if (!emailRegex.test(email)) {
+      EsetError('Invalid email address.');
+      setShakeEmail(true);
+      valid = false;
+    }
+
+    // Validate password
+    if (!password) {
+      PsetError('Password cannot be empty.');
+      setShakePassword(true);
+      valid = false;
+    } else {
+      let errors = [];
+
+      if (!minLengthRegex.test(password)) {
+        errors.push('* must contain min. 8 characters');
+      }
+      if (!containsLetterRegex.test(password)) {
+        errors.push('* must contain min. 1 letter');
+      }
+      if (!containsNumberRegex.test(password)) {
+        errors.push('* must contain min. 1 number');
+      }
+      if (!containsSpecialCharRegex.test(password)) {
+        errors.push('* must contain min. 1 special character');
+      }
+      if (errors.length > 0) {
+        PsetError(errors.join('\n'));
+        setShakePassword(true);
+        valid = false;
+      }
+    }
+
+    // Validate confirm password
+    if (!confirmpassword) {
+      confirmPsetError('Please confirm password.');
+      setShakeConfirmPassword(true);
+      valid = false;
+    } else if (password !== confirmpassword) {
+      confirmPsetError('Passwords does not match.');
+      setShakeConfirmPassword(true);
+      valid = false;
+    }
+
+    if (!valid) {
+      setTimeout(() => {
+        setShakeEmail(false);
+        setShakePassword(false);
+        setShakeConfirmPassword(false);
+      }, 100);
+      return;
+    }
+  };
+
+  return (
+    <main className={styles.SignUpPageContainer}>
```

Comment:
outer element for a component should always either be `<></>` which is called a react fragment, or `<div></div>`. <main> is typically only used in the index page.